### PR TITLE
axios

### PR DESCRIPTION
**Pull request by thapar-vansh**

### CODE REVIEW

File: controllers/axiosController.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import axios from 'axios'
+import { Request, Response } from 'express'
+
+export const getRequest = async (
+  req: Request,
+  res: Response
+): Promise<Response> => {
+  const response = await axios.get('https://jsonplaceholder.typicode.com/posts')
```
Comment (Sachin-R-Yawalkar) [inline controllers/axiosController.ts]: Action: add axios request in try catch block

File: controllers/axiosController.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  return res.send(response.data)
+}
+
+export const postRequest = async (req: Request, res: Response) => {
+  try {
+    const data = req.body
+    await axios.post('https://jsonplaceholder.typicode.com/posts', data)
```
Comment (Sachin-R-Yawalkar) [inline controllers/axiosController.ts]: What will happen if this fails?

File: controllers/axiosController.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    return res.send('Added post')
+  } catch (error) {
+    console.log(error)
+  }
+}
+
+export const putRequest = async (req: Request, res: Response) => {
+  try {
+    const data = req.body
+    await axios.post('https://jsonplaceholder.typicode.com/posts', data)
```
Comment (Sachin-R-Yawalkar) [inline controllers/axiosController.ts]: What will happen if this fails?

File: controllers/axiosController.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    return res.send('Updated post')
+  } catch (error) {
+    console.log(error)
+  }
+}
+
+export const deleteRequest = async (req: Request, res: Response) => {
+  try {
+    await axios.delete('https://jsonplaceholder.typicode.com/posts/1')
```
Comment (Sachin-R-Yawalkar) [inline controllers/axiosController.ts]: What will happen if this fails?

**Review by Sachin-R-Yawalkar — CHANGES_REQUESTED** [review]

File: controllers/axiosController.ts [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  }
+}
+
+export const postRequest = async (
+  req: Request,
+  res: Response
+): Promise<Response> => {
+  try {
+    const data = req.body
+    await axios.post('https://jsonplaceholder.typicode.com/posts', data)
```
Comment (jagdishcm-sixer) [inline controllers/axiosController.ts]: You can put the url in some common file like constants or env file.

Comment (jagdishcm-sixer) [inline controllers/axiosController.ts]: You can add one more check for response code if it's 200 or not.

**Review by jagdishcm-sixer — CHANGES_REQUESTED** [review]

**Review by jagdishcm-sixer — APPROVED** [review]

**Review by Sachin-R-Yawalkar — APPROVED** [review]

**Pull request merged**